route-rs-runtime/src/packet/ipv4.rs

```diff
@@ -0,0 +1,72 @@
+use crate::packet::*;
+
+
+pub struct Ipv4Packet<'packet> {
+    pub data: PacketData<'packet>,
+}
+
+impl<'packet> Ipv4Packet<'packet> {
+    fn new(packet: PacketData) -> Result<Ipv4Packet, &'static str> {
+        //Header of Ethernet Frame: 14 bytes
+        //Header of IPv4 Frame: 20 bytes
+        if packet.len() < 14 + 20 {
```

Comment:
Ideally a packet type doesn't have to worry about what its containing packet type is. This is worth discussion.